### PR TITLE
Feature streering platform

### PR DESCRIPTION
**Pull request by marcelkuipers**
Unintentionally I misspelled the name of this branch.
You can now steer the platform by pressing left and right.
It has been implemented such that you can easily make steering more gradual by changing a float value. Also, the maximum angle under which you can steer can also be changed.

### CODE REVIEW

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 1]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
                          default: //Do nothing when an unknown button is pressed.
                      }
                  }
+                 if(keyPressed){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: if should be followed by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: { is not preceded by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 2]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
...
+                        if(name.equals("steer_left")){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: if should be followed by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: { is not preceded by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 3]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
...
+                        environmentState.steer(-1.f);
+                    }
+                    if(name.equals("steer_right")){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: if should be followed by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: { is not preceded by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 4]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
...
+                        environmentState.steer(1.f);
+                    }
+                 }
+                 if(!(keyPressed)){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: if should be followed by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: { is not preceded by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: unnecessary bracket around keyPressed.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 5]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
...
+                        if(name.equals("steer_left") || name.equals("steer_right")){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: if should be followed by a whitespace.

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java]: CheckStyle: { is not preceded by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 6]
```diff
@@ -140,6 +140,22 @@ public void onAction(String name, boolean keyPressed, float tpf) {
...
+                        environmentState.steer(0.f);
+                    }
+                    
+                 }
+                 
+
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java:144]: Can you move this into a separate private method and then call that method here?
This initInputs method is long as it is.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 7]
```diff
@@ -101,8 +104,11 @@ public void initialize(AppStateManager stateManager, Application app) {
     @Override
     public void update(float tpf) {
         super.update(tpf);
-        testPlatform.move(-PLATFORM_SPEED, 0, 0);
-        testCommander.move(-PLATFORM_SPEED, 0, 0);
+        float zAxis = (float) (steering*(Math.sqrt(2.f)/2.f));
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: CheckStyle:
`*` should be preceded and followed by a whitespace.
`/` should be preceded and followed by a whitespace.

I also think the brackets around `steering*(Math.sqrt(2.f)/2.f)` are not needed

2.f Feels like a magic number

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 8]
```diff
@@ -101,8 +104,11 @@ public void initialize(AppStateManager stateManager, Application app) {
...
+        float xAxis = (float) Math.sqrt(1-Math.pow(zAxis,2));
+        
+        testPlatform.move(-PLATFORM_SPEED*xAxis, 0, -PLATFORM_SPEED*zAxis);
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: Checkstyle: `*` should be preceded and followed by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 9]
```diff
@@ -101,8 +104,11 @@ public void initialize(AppStateManager stateManager, Application app) {
...
+        testCommander.move(-PLATFORM_SPEED*xAxis, 0, -PLATFORM_SPEED*zAxis);
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: Checkstyle: `*` should be preceded and followed by a whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 10]
```diff
@@ -296,5 +275,41 @@ private void updateTestWorld() {
 
     }
 
+    /**
+     * Update the steering direction.
+     * @param orientation
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: CheckStyle: missing param description.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 11]
```diff
@@ -296,5 +275,41 @@ private void updateTestWorld() {
 
     }
 
+    /**
+     * Update the steering direction.
+     * @param orientation 
+     */
+    public void steer(float orientation){
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: CheckStyle: { should be preceded by whitespace.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 12]
```diff
@@ -296,5 +275,41 @@ private void updateTestWorld() {
...
+    }
+
+
+    /**
+     * Handles everything that happens when the Envrironment state is detached from the main application.
+     */
+    @Override
+    public void cleanup() {
+        super.cleanup();
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java:295]: Why do you need to override the super cleanup method if that's the only thing you're calling?

**Comment by marcelkuipers** [comment]
I am fixing it right now.

**Comment by marcelkuipers** [comment]
I fixed the checkstyle errors and implemented the recommendations.

**Comment by NilsHullegien** [comment]
Code looks good, we might want to tweak the values for the steering later when the map is done.

**Comment by LMiljak** [comment]
It's good now! Merging.

**Pull request merged**